jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -934,7 +934,11 @@ public void onFlushed(long bytes) throws IOException
         if (LOG.isDebugEnabled())
             LOG.debug("Flushed bytes min/actual {}/{}", minFlushed, _flushed);
         if (_flushed < minFlushed)
-            throw new IOException(String.format("Response content data rate < %d B/s", minDataRate));
+        {
+            IOException ioe = new IOException(String.format("Response content data rate < %d B/s", minDataRate));
```

Comment:
Why do we throw a `BadMessageException` for the input and an `IOException` for the output?

Reply:
I think it's because `BadMessageException` is a special code to possibly trigger a `400 Bad Message` response.

The existing APIs for output (`HttpServletResponse`) only have `IOException`.

Reply:
slow input is seen as a bad message from the client which should solicit a 408 response (if we were willing to no abort).
slow output is a client being slow reading a good message from the server.

I guess they both could be IOExceptions and we could just set a 408 for the request log?

Reply:
@gregw I would like that better: i.e. abort slow input with a 408 to feed the request log, but throw an `IOException` to applications like we do for slow output. As we have already aborted, there is nothing an application will be able to do, so `IOException` looks more fatal than a `BadMessageException`.
Consider if it needs to be a Jetty `QuietException` (e.g. `EofException`) so that it does not get logged over and over at WARN level. Perhaps introduce a `QuietIOException`?

Reply:
Wouldn't it be fatal to not finish a java Stream normally, but with an exception? (regardless of read/write)

Reply:
aborting is fatal for the connection, the exception is just to tell the application that it is over!
I'm now throwing IOException in HttpInput as well